compiler/damlc/daml-lf-conversion/src/DA/Daml/LFConversion.hs

```diff
@@ -968,25 +1000,24 @@ useSingleMethodDict env (Cast ghcExpr _) f = do
 useSingleMethodDict env x _ =
     unhandled "useSingleMethodDict: not a single method type class dictionary" x
 
-convertImplements :: Env -> LF.TypeConName -> ConvertM (NM.NameMap TemplateImplements)
+convertImplements :: Env -> Qualified LF.TypeConName -> ConvertM (NM.NameMap TemplateImplements)
 convertImplements env tpl = NM.fromList <$>
-  mapM convertInterface (MS.findWithDefault [] tpl (envImplements env))
+  mapM convertInterface (MS.findWithDefault [] (qualObject tpl) (envImplements env))
   where
     convertInterface :: (Maybe LF.SourceLoc, GHC.TyCon) -> ConvertM TemplateImplements
     convertInterface (originLoc, iface) = withRange originLoc $ do
       let handleIsNotInterface tyCon =
             "cannot implement '" ++ prettyPrint tyCon ++ "' because it is not an interface"
-      con <- convertInterfaceTyCon env handleIsNotInterface iface
-      let mod = nameModule (getName iface)
+      ifaceCon <- convertInterfaceTyCon env handleIsNotInterface iface
 
       -- TODO: Stub view, will extract later, https://github.com/digital-asset/daml/pull/14439
-      let view = ETmLam (ExprVarName "this", TCon con) (EBuiltin BEUnit)
+      let view = ETmLam (ExprVarName "this", TCon ifaceCon) (EBuiltin BEUnit)
 
       methods <- convertMethods $ MS.findWithDefault []
-        (mod, qualObject con, tpl)
+        (tpl, ifaceCon)
```

Comment:
I also reordered this from `(~interface module~, interface tycon, template tycon)` to `(template qtycon, interface qtycon)` to match the order in the `ImplementsT` markers